### DISCUSSION TITLE
Setting up a simple layout reset

### DISCUSSION THREAD

**Discussion by hubert-ix** [sveltejs/kit#6445] [Q&A]
I've just updated to the latest version of SvelteKit. I was on 1.0.0-next.405, the last version before all the massive route changes, and my (sizeable) app is working fine on that version.

After running the migration tool and launching "npm run dev", the first error I get is for a reset layout I have somewhere in a route.

My layouts are pretty simple: I have one ```+layout.svelte``` at the root, and sometimes I need to completely reset it so that there's just a ```<slot />```

The docs say: 

> For example, a ```+layout@.svelte``` component would reset the hierarchy for all its child routes

So I've put a ```+layout@.svelte``` file (with just ```<slot />```) where I want the layout to reset, but it's telling me:

_my-route/+layout@.svelte references missing layout "_

How shall I do a simple layout reset then?

**Comment by paulwvnjohi** [comment]
you can used groupings, https://kit.svelte.dev/docs/advanced-routing#advanced-layouts-group

**Comment by coldino** [comment]
I'm in the same boat. I understand I can use groups as an alternative, but shouldn't this work as-is?

The docs say `+layout@.svelte` should do what I expect, and yet I cannot use it at all due to the same error as above.

**Comment by matiadev** [comment]
To reset a layout create a blank root layout `+layout.svelte` and then you can reference it using `+layout@.svelte`.

```
.
├── page/
│   ├── +page.svelte
│   └── +layout@.svelte
└── +layout.svelte
```

**Reply by coldino** [reply]
Thanks, but it's looking like this wasn't the problem. A fresh SvelteKit project works using `+layout@.svelte` even without an empty layout at the root.

For me, I didn't realise the automated migration to the new router wouldn't update the packages - so I ended up with a mixture of old/new design. Fixing this up has removed the error.

**Reply by louis-bourgault** [reply]
Thank you, that worked really well for me and so simple

**Comment by scottBowles** [comment]
Maybe I'm severely missing something, but I really can't figure out how to reset the layout.

I _think_ I'm doing as suggested to remove the layout from the `About` page here: https://stackblitz.com/edit/sveltejs-kit-template-default-dnvpmm?file=src/routes/(about-group)/about/+page.svelte

I've also tried with the `@` following the other layout file in the `about-group` folder, and with `@` followed by various things in `about/page.js` (e.g., `about/page@about.js`).

**Reply by dummdidumm** [reply]
`+layout@.svelte` or `+page@.svelte` means you reset to the root layout. The root layout is `src/routes/+layout.svelte`. In your example the root layout contains all the styling and the layouts below nothing, so resetting doesn't change anything UI-wise.

**Reply by hubert-ix** [reply]
Ok that makes sense now. But then it means you need to have a blank layout as your root layout, which in practice will almost never happens.

**Reply by scottBowles** [reply]
@dummdidumm Thanks for your help. This might be one of those things that seems obvious once you see it but isn't at first, or maybe it was just me. I'm with @hubert-ix that it feels a little clunky. Might be worth mentioning in the docs with an example that you'll need a blank layout at root if you want to ever be able to reset. Or maybe worth letting `+layout@.svelte` rewind to a blank layout if there isn't a `+layout.svelte` at root to avoid the boilerplate.

**Reply by dummdidumm** [reply]
I can understand that it's weird to have a blank root layout reserved for this. On the other hand, you are guaranteed that this root layout _always_ exists so you can put app-wide behavior in there. Previously this wasn't possible and resetting to a layout outside the chain caused all kinds of subtle inconsistencies and confusion. The docs could make this more clear, agree on that.

**Reply by hubert-ix** [reply]
So if I am getting this right, the solution is:
- create a blank root layout
- add a group folder (something like ```(app)```) and put all your route folders inside it (so that we don't have to change all the route's URLs)
- put the proper layout in the ```(app)``` folder
- whenever you need to reset the layout in a folder, use ```+layout@.svelte```, which will reset to the blank root layout

**Reply by yousufiqbal** [reply]
This looks _overkill_ for a simple reset. I'm getting these vibes for first time in SvelteKit. There should be a simpler solution. One should be able to live without groups and apply a reset as well.

**Reply by yousufiqbal** [reply]
Maybe a `+layout.reset.svelte`?

**Reply by hubert-ix** [reply]
@yousufiqbal I'm with you. I'm all for embracing change, but in practice the recent updates in sveltekit have proved to add a lot more work rather than simplifying things. At least that's been my experience so far.

**Reply by stephane-vanraes** [reply]
I fell back to the "old style" of checking the current route in an `if` clause in my root layout to differentiate when a page needs a new layout at that level.  It adds a touch more code, but at least my route files are together instead of in two different subtrees.

While I understand the logic behind the "grouped layouts" to a point, it's a pity named layouts have been removed.  Would like to see those making a comeback.

**Reply by Greenheart** [reply]
Thanks @stephane-vanraes! Your suggestion worked great - wrapping the root `+layout.svelte` in a conditional to render an empty layout for the routes that need it.

A pretty clean solution, and only minimal changes needed to make it work.

Layout groups and other advanced features are likely useful in larger projects, but for a simple static site, the if block is as simple as it can get. No need to introduce SvelteKit-specific directory structures and filenames when an if statement is enough :)
